### ISSUE TITLE
`--exitonerror` should cause a non-zero exit code

### ISSUE THREAD

**Issue by DetachHead** [robotframework/robotframework#4853]
currently it only fails remaining tests if an error occurs, but if for example an error occurs in a listener's `end_test` method, the test will have already passed and the exit code will be 0 if there are no more tests to run

**Comment by pekkaklarck** [comment]
I agree the return code should always be non-zero if execution ends for an error. Unfortunately handling `--exit-on-error` is somewhat tricky and this may not be that easy to implement. Let's try to come up with some solution in RF 7.1. If it's too complicated, we can latest fix this when #4170 that is also blocked by problems caused by `--exit-on-error` is implemented.

**Comment by DetachHead** [comment]
this also occurs if an import fails, which is a much more common case

**Comment by pekkaklarck** [comment]
I looked at this and #4170 again and required changes are too big for RF 7.2.
